### PR TITLE
SW-2283 Add ecosystem types to species

### PR DESCRIPTION
**Pull request by sgrimm**
A species can now have a list of ecosystem types. The list of types is included 
in the species API payloads and is available via search.

It must also be included in CSVs; if a species has more than one ecosystem type,
the types should be on separate lines in single cell of the CSV.

### CODE REVIEW

**Comment by sgrimm** [comment]
Current dependencies on/for this PR:
* main
  * **PR #921** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/921" target="_blank">Graphite</a> 
    * **PR #868** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/868" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/terraware/terraware-server/868?utm_source=stack-comment).

**Comment by sgrimm** [comment]
Marked this as a draft pending answers to some questions on the Jira task; this PR implements what I think the answers are most likely to be.

**Comment by sgrimm** [comment]
Renamed "ecosystem" to "biome" per Jira task discussion and added it to the species list import CSV format.

**Comment by sgrimm** [comment]
@charlie-tf This isn't quite finished yet but wanted to update the draft so you can see what the API changes will look like. There's a new optional field `ecosystemTypes` on the species payloads and it's a list of ecosystem types (see the change to `R__TypeCodes.sql` for the list of values, or you can pull the branch and generate the OpenAPI schema).

I expect to have this wrapped up tomorrow but didn't want you to be blocked waiting for the API schema.

**Comment by sgrimm** [comment]
This should be ready to go now.

File: src/main/resources/i18n/Messages.properties
Comment (charlie-tf) [inline src/main/resources/i18n/Messages.properties]: "`...list of any of:`"? I'm not sure, it's hard to not make this wording a bit awkward.

File: src/main/kotlin/com/terraformation/backend/species/db/SpeciesCsvValidator.kt
Comment (charlie-tf) [inline src/main/kotlin/com/terraformation/backend/species/db/SpeciesCsvValidator.kt:20]: Does this expect the delimiter between ecosystems to be a newline/cr, possibly preceded or followed by whitespace (sorry i'm kinda bad at reading regex 😅 )?

Comment (charlie-tf) [reply src/main/kotlin/com/terraformation/backend/species/db/SpeciesCsvValidator.kt:20]: Oh, looks like it https://github.com/terraware/terraware-server/pull/868/files#diff-01b6bb49f729754a9021737e7cf0f0f28ed934f08ec7feb22b1011c6505ce34fR344 feels a little odd for csv (rows with embedded newlines, that is). Is the parser not confused as it is consuming the file line-by-line?

**Review by charlie-tf — COMMENTED** [review]
Seems mostly good! I just had a question about the importer and a wording choice.

File: src/main/kotlin/com/terraformation/backend/species/db/SpeciesCsvValidator.kt
Comment (sgrimm) [reply src/main/kotlin/com/terraformation/backend/species/db/SpeciesCsvValidator.kt:20]: The CSV parser isn't strictly line-based; it's smart enough to know that quoted cell values can include line breaks.

File: src/main/resources/i18n/Messages.properties
Comment (sgrimm) [reply src/main/resources/i18n/Messages.properties]: Oops, thanks for catching that. I meant to write what you wrote. I was playing with this to try to find a non-awkward wording.

File: src/main/kotlin/com/terraformation/backend/search/table/SpeciesEcosystemTypesTable.kt
Comment (charlie-tf) [inline src/main/kotlin/com/terraformation/backend/search/table/SpeciesEcosystemTypesTable.kt:33]: 👍

**Review by charlie-tf — APPROVED** [review]
Looks good! Sorry Steve, I didn't realize this was still waiting for an approval.

**Pull request merged**
